### ISSUE TITLE
Updating the inventory seems broken

### ISSUE THREAD

**Issue by mordante** [Brewtarget/brewtarget#315]
When updating the amount of ingredients I have in my inventory it seems to behave odd. Most of the time an update is ignored. I analysed the bug and think I found a solution, but I dare not to invoke the wrath of the ancient ones on my database.

I think this untested patch should do the trick:
```
diff --git a/src/database.cpp b/src/database.cpp
index 57194d1..8fa74ea 100644
--- a/src/database.cpp
+++ b/src/database.cpp
@@ -1594,7 +1594,7 @@ void Database::updateEntry( Brewtarget::DBTable table, int key, const char* col_
       QString command = QString("UPDATE %1 set %2=:value where id=%3")
                            .arg(tableName)
                            .arg(col_name)
-                           .arg(object->_key);
+                           .arg(key);

```

When I edit the inventory entry with id 9 the non-existing entry with id 142 is updated instead. This can be seen in the following part of my GDB session, while changing the amount of hops:
```
(gdb) f 0
#0  Database::updateEntry (this=0xd51970, table=Brewtarget::HOPINVTABLE, key=9, col_name=0x78c798 "amount", value=..., prop=..., object=0xdc1530, notify=true, 
    transact=false) at /data/src/brewtarget/src/database.cpp:1599
1599	ate.prepare( command );
(gdb) p *object
$35 = {
  <QObject> = {<No data fields>}, 
  members of BeerXMLElement: 
  static staticMetaObject = {
    d = {
      superdata = 0x7ffff5ec0be0 <QObject::staticMetaObject>, 
      stringdata = 0x80bb40 <qt_meta_stringdata_BeerXMLElement>, 
      data = 0x80bd80 <qt_meta_data_BeerXMLElement>, 
      static_metacall = 0x7584a4 <BeerXMLElement::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, 
      relatedMetaObjects = 0xaaeab0 <qt_meta_extradata_BeerXMLElement>, 
      extradata = 0x0
    }
  }, 
  _key = 142, 
  _table = Brewtarget::HOPTABLE, 
  _valid = true, 
  _folder = {
    static null = {<No data fields>}, 
    d = 0x7ffff5b1fcc0
  }, 
```

The table and key (9) send as arguments are the correct ones, but the key of the object (142) is used to update the table.

If you want more information feel free to ask. I've also joined the IRC chan.

**Comment by mikfire** [comment]
Nicely done. Your diagnosis was correct, and your patch was correct as well. As soon as the travis CI finishes, I will merge the patch in.

If you want to continue contributing (please?), I would suggest at least creating a development user so you don't mess your own database up. I actually run several VMs to keep everything as separate as I can. I also run three different postgreSQL databases so I can play with those as I need/want.

**Comment by mikfire** [comment]
Closed with pr #317
